main.tf

```diff
@@ -1,3 +1,324 @@
-resource "random_pet" "template" {
-  length = var.length
+# This Terraform code automates the creation and management of Spacelift stacks based on the structure
+# and configurations defined in the Git repository, default Stack values and additional input variables.
+# It primarily relies on dynamic local expressions to generate configurations based on the
+# input variables and Git structure so it can be consumed by Spacelift resources.
+# This module can also manage the automation stack itself, but it should be bootstrapped manually.
+#
+# It handles the following:
+#
+# 1. Stack Configurations (see ## Stack Configurations)
+# Reads the Spacelift stack configurations strictly based on the root modules structure in Git and file names.
+# These are the configurations required to be set for a stack, e.g. project_root, terraform_workspace, root_module.
+#
+# 2. Common Stack configurations (see ## Common Stack configurations)
+# Some configurations are equal across the whole root module, and can be set it on a root module level:
+# * Space IDs: in the majority of cases all the workspaces in a root module belong to the same Spacelift space, so
+# we allow setting a "global" space_id for all stacks on a root module level.
+# * Autodeploy: if all the stacks in a root module should be autodeployed.
+# * Administrative: if all the stacks in a root module are administrative, e.g stacks that manage Spacelift resources.
+#
+# 3. Labels (see ## Labels)
+# Generates labels for the stacks based on administrative, dependency, and folder information.
+#
+# Syntax note:
+# The local expression started with an underscore `_` is used to store intermediate values
+# that are not directly used in the resource creation.
+
+locals {
+  enabled = module.this.enabled
```

Comment:
If we're only include context for the purpose of `this.enabled`... we may want to remove it to simplify this example and not confuse people.

Reply:
Is there anywhere we could be using tags? Do we even supply any of the naming components to this module? My thought is that we remove...

Reply:
No, Spacelift TF resources don't have tags attribute. I agree - it doesn't help much in this case.